nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/planner_selector_node.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Pablo Iñigo Blasco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__PLANNER_SELECTOR_NODE_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__PLANNER_SELECTOR_NODE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "std_msgs/msg/string.hpp"
+
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_behavior_tree
+{
+
+class PlannerSelector : public BT::DecoratorNode
```

Comment:
@naiveHobo does this make most sense as a decorator or an action node?

Reply:
I think making it an action node makes more sense. As a decorator, it's not really affecting *how* its child node is ticked. The main task this node is supposed to perform (only setting something on the blackboard) is similar to the [SetBlackboard node](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/master/include/behaviortree_cpp_v3/actions/set_blackboard_node.h) which is also an action node.

Reply:
Yeah, I know he was following the template of the goal updater node, which I also think is a bit odd to be a decorator

Reply:
I asked about the question of the decorator or the action node in [the discussion issue 147](https://github.com/ros-planning/navigation.ros.org/issues/147#issuecomment-799728405). It is better think before code :-).

I will change it to an bt-action-node.

Reply:
Sorry, that got lost in the details I was thinking about in that comment! I also don't think there's an objectively right answer to this question

Reply:
No problem, please, just confirm me if this must be done. y/N?

Reply:
Yes, make it an action node, but you don't need the https://github.com/ros-planning/navigation2/blob/main/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L30 BTActionNode object, since this isn't also a ROS action. I think you just need to use the BT.CPP SyncActionNode (or even `ActionNodeBase`). A simple example is [here](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/master/include/behaviortree_cpp_v3/actions/set_blackboard_node.h)

Reply:
Correct, I see BTActionNodes are for action clients, so that does not apply our case.
I will check those base nodes you say.